stream/src/main/java/org/apache/apex/malhar/stream/api/WindowedStream.java

```diff
@@ -51,6 +67,140 @@
+   * Also give a name to the accumulation
+   * @param accumulation Accumulation function you want to do
+   * @param <O> The output type that you want to accumulate the value to
+   * @param <ACCU> The type of accumulation you want to keep (it can be in memory or on disk)
+   * @param <STREAM> return type
+   * @return
+   */
+  <O, ACCU, STREAM extends WindowedStream<Tuple.WindowedTuple<O>>> STREAM accumulate(Accumulation<T, ACCU, O> accumulation, Option... opts);
+
+  /**
+   * Add {@link WindowedOperatorImpl} with specified {@link ReduceFn} <br>
+   * Do reduce transformation<br>
+   * @param reduce reduce function
+   * @param <STREAM> return type
+   * @return new stream of same type
+   */
+  <STREAM extends WindowedStream<Tuple.WindowedTuple<T>>> STREAM reduce(ReduceFn<T> reduce, Option... opts);
+
+  /**
+   * Add {@link KeyedWindowedOperatorImpl} with specified {@link ReduceFn} <br>
+   * Reduce transformation by selected key <br>
+   * Add an operator to the DAG which merge tuple t1, t2 to new tuple by key
+   * @param reduce reduce function
+   * @param convertToKeyVal The function convert plain tuple to k,v pair
+   * @param <K> The type of key you want to group tuples by
+   * @param <V> The type of value extract from tuple T
+   * @param <STREAM> return type
+   * @return new stream of key value pair
+   */
+  <K, V, STREAM extends WindowedStream<Tuple.WindowedTuple<KeyValPair<K, V>>>> STREAM reduceByKey(ReduceFn<V> reduce, Function.ToKeyValue<T, K, V> convertToKeyVal, Option... opts);
+
+
+  /**
+   * Add {@link WindowedOperatorImpl} with specified {@link FoldFn} <br>
+   * Fold transformation <br>
+   * @param fold fold function
+   * @param <O> output type of fold function
+   * @param <STREAM> return type
+   * @return
+   */
+  <O, STREAM extends WindowedStream<Tuple.WindowedTuple<O>>> STREAM fold(FoldFn<T, O> fold, Option... opts);
+
+  /**
+   * Add {@link KeyedWindowedOperatorImpl} with specified {@link FoldFn} <br>
+   * Fold transformation by key <br>
+   * @param fold fold function
+   * @param <O> Result type
+   * @return new stream of type O
+   */
+  <K, V, O, STREAM extends WindowedStream<Tuple.WindowedTuple<KeyValPair<K, O>>>> STREAM foldByKey(FoldFn<V, O> fold, Function.ToKeyValue<T, K, V> convertToKeyVal, Option... opts);
+
+
+  /**
+   * Return tuples for each key for each window
+   * @param <O>
+   * @param <K>
+   * @param <STREAM>
+   * @return
+   */
+  <O, K, STREAM extends WindowedStream<KeyValPair<K, Iterable<O>>>> STREAM groupByKey(Function.ToKeyValue<T, K, O> convertToKeyVal, Option... opts);
+
+  /**
+   * Return tuples for each window
+   * @param <STREAM>
+   * @return
+   */
+  <STREAM extends WindowedStream<Iterable<T>>> STREAM group();
+
+  /**
+   * Reset the trigger settings for next transforms
+   * @param triggerOption
+   * @param <STREAM>
+   */
+  <STREAM extends WindowedStream<T>> STREAM resetTrigger(TriggerOption triggerOption);
```

Comment:
You specify the first trigger for WindowedStream when you convert a normal stream to a WindowedStream using window method. After that if you want to set next transformation with different trigger call this method.